### DISCUSSION TITLE
How long until v1. 0?

### DISCUSSION THREAD

**Discussion by txtyash** [leptos-rs/leptos#2090] [Q&A] [answered]
This is an unhealthy and a stupid question.

I have tried out leptos and I love it. I want to start a big project in leptos but I also wished it were stable. I hate to use javascript. I am not in a rush to start the project. I wanted to know how long do you guys think until leptos is stable?

*I do not intend to cause any disruption or doubt the speed of development. I love leptos and just wanted to ask this question.*

**Comment by gbj** [comment, marked as answer]
1. Thank you for the grace and humility with which you asked this question.
2. Be kind to yourself. It is not at all stupid or unhealthy, and I appreciate the spirit in which you're asking it.
3. Most of what's in my note about being "production ready" in the README is true ([link](https://github.com/leptos-rs/leptos#is-it-production-ready) if you haven't seen it)
4. I'm in the middle of a pretty fundamental rewrite. It is roughly on par with the scale of the change from 0.0 to 0.1, i.e., most of the core of the library is being rewritten. However, my goal is for most of the internals to be rewritten and 90-95% of actual application code (all your reactive stuff, view macro, component macros, etc.) to remain the same or be "search and replace" changes on function names etc. There will be some nice benefits (better `async` support, smaller binary sizes, faster HTML rendering, better type system support for checking HTML attribute names etc.) but may be some things that used to work that don't work any more (storing `View`s in signals and passing them around would be one example)
5. The further I get into the rewrite, the more convinced I am that it's the best Rust code I've written, enables much more idiomatic Rust in the library, and is much better than the somewhat-tangled mess of the current codebase. I also think it's fundamentally the best approach for doing what we're doing.

So, say I release 0.6 in the next 4-6 months. I'd probably like it to be stable for at least another 6-12 months of real use without needing any breaking changes before I'd release it as 1.0. But my work on 0.6 feels to me like the first version of the library that would be a possible 1.0.

In the meantime, there are people who are in fact using Leptos in prod with success (https://www.houski.ca/, https://www.rustadventure.dev/ come to mind), so don't feel the need to wait for 1.0 to get started...

I hope that helps!

**Reply by gbj** [reply]
Oh, for the sake of clarity I should add: I tend to use semantic versioning in the way it's approached in the Rust community, not the JS community. So "1.0" is a pretty specific guarantee about long-term API stability, not about production-readiness or general feature-stability. A version number over 2 signifies either "something went wrong here" or "there was a major paradigm shift in the domain this library works in." One of the things I love so much about Rust is that the code I wrong with `serde = "1"` three years ago still compiles and works exactly the same when I use it in an application that uses `serde = "1"` now. That kind of long-term stability is what 1.0 means to me.

1.0 in this sense is also quite hard before other dependencies go to 1.0. For example, `leptos_axum` 1.0 probably can't come before `axum` 1.0, because when `axum` goes to 1.0 we'd have to release `leptos_axum` 2.0 due to breaking changes, etc.
